modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/dsl/metrics/Metrics.scala

```diff
@@ -150,8 +157,24 @@ object Metrics {
 
   sealed trait KVMetrics {
     def toList: List[KVMetric] = this match {
-      case KVMetrics.LoadingCompleted(countGood, countBad, minTstamp, maxTstamp, shredderStart, shredderEnd) =>
-        List(Some(countGood), Some(countBad), minTstamp, maxTstamp, Some(shredderStart), Some(shredderEnd)).unite
+      case KVMetrics.LoadingCompleted(
+            countGood,
+            countBad,
+            minTstamp,
+            maxTstamp,
+            shredderStart,
+            shredderEnd,
+            recoveryTablesLoaded
+          ) =>
+        List(
+          Some(countGood),
+          Some(countBad),
+          minTstamp,
+          maxTstamp,
+          Some(shredderStart),
+          Some(shredderEnd),
+          Some(recoveryTablesLoaded)
```

Comment:
Wonder if we should omit (not emit) the metric if it is zero?  Or if we're loading to snowflake/databricks for which it has no meaning.

Reply:
I think this question depends on the downstream dependencies - we have statsD, then we have the metrics relay/custom tooling to push that data to eg. cloudwatch. I'm not sure if I remember this well enough but I think some of those things can fail if they expect a metric that isn't present - but I might be wrong.

I think my opinion stands whatever the answer to the above consideration is though:

- For Snowflake/Databricks, I would omit it, assuming the tf stacks are sufficiently separated that doing so is trivial and doesn't introduce complexity.
- For Redshift I would not omit it, even if there were no recovery tables produced - in the case of some confusing issue, knowing that 0 were produced can be as valuable as knowing that 1 or more were produced. I encountered such a scenario recently in QA.

Reply:
I just pushed an extra commit, which disables the metric for Snowflake/Databricks loaders.